src/controllers/mealController.ts

```diff
@@ -1,20 +1,107 @@
 
 export const getAllMeals = async (req: Request, res: Response) => {
-    const { ingredients } = req.query;
-    if (!ingredients || typeof ingredients !== 'string') {
-        throw new BadRequestError('Please provide a list of ingredients');
-    } else {
-        try {
-            const recipes = await getAllMealsAPI(ingredients.split(','))
-            return res.json(recipes).status(StatusCodes.OK)
-        } catch (error) {
-            // return res.status(500).json({ message: `${error}` })
-            throw new ServerError(`${error}`)
-        }
+  const { ingredients } = req.query;
+  if (!ingredients || typeof ingredients !== "string") {
+    throw new BadRequestError("Please provide a list of ingredients");
+  } else {
+    try {
+      const recipes = await getAllMealsAPI(ingredients.split(","));
+      return res.json(recipes).status(StatusCodes.OK);
+    } catch (error) {
+      // return res.status(500).json({ message: `${error}` })
+      throw new ServerError(`${error}`);
     }
-}
\ No newline at end of file
+  }
+};
+
+export const getRandomMealsUnauthenticated = async (
+  req: Request,
+  res: Response,
+) => {
+  try {
+    const { size } = req.body;
+    const queryRange = size ? parseInt(size.toString()) : 30;
+
+    const response: { randomMeals: {}[] } = {
+      randomMeals: [],
+    };
+    const uniqueCheck = new Set<string>([]);
+
+    for (let i = 0; i < queryRange; i++) {
+      let randomMeal = await getRandomMeal();
+      // check for duplicate meals
+      while (randomMeal.strMeal && uniqueCheck.has(randomMeal.strMeal)) {
+        randomMeal = await getRandomMeal();
+      }
+      uniqueCheck.add(randomMeal.strMeal);
+      response.randomMeals.push(randomMeal);
+    }
+
+    return res.json(response).status(StatusCodes.OK);
+  } catch (error) {
+    throw new ServerError(`${error}`);
+  }
+};
+
+export const getRanDomMealsAuthenticated = async (
+  req: Request,
+  res: Response,
+) => {
+  try {
+    const { mainSize, sideSize, dessertSize } = req.body;
+    const mainRange = mainSize ? parseInt(mainSize.toString()) : 10;
+    const sideRange = sideSize ? parseInt(sideSize.toString()) : 5;
+    const dessertRange = dessertSize ? parseInt(dessertSize.toString()) : 5;
+
+    const response: {
+      mainMeals: {}[];
+      sideMeals: {}[];
+      dessertMeals: {}[];
+    } = { mainMeals: [], sideMeals: [], dessertMeals: [] };
+
+    response.sideMeals = await getMealByFilter("category", "Side", sideRange);
+    response.mainMeals = await getMealByFilter(
+      "category",
+      getRandomKey(MainCategories),
+      mainRange,
+    );
+    response.dessertMeals = await getMealByFilter(
+      "category",
+      "Dessert",
+      dessertRange,
+    );
+
+    return res.json(response).status(StatusCodes.OK);
+  } catch (error) {
+    throw new ServerError(`${error}`);
+  }
+};
+
+export const getMealsFromLeftOver = async (req: Request, res: Response) => {
+  try {
+    let { ingredients } = req.body;
```

Comment:
> Change this to mock data. Or if you want, looking at the Usermodels, there is a field left over which is an array of objectid. Populate it then connect at here. We don't send the data from frontend.

Hey Hung I changed the function so that now we will get the ingredients from the leftOver field of a mock user document. Here's a sample data 

```json
{
    "suggestedMeals": [
        [
            {
                "strMeal": "Apple & Blackberry Crumble",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/xvsurr1511719182.jpg",
                "idMeal": "52893"
            },
            {
                "strMeal": "Bread and Butter Pudding",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/xqwwpy1483908697.jpg",
                "idMeal": "52792"
            },
            {
                "strMeal": "Budino Di Ricotta",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/1549542877.jpg",
                "idMeal": "52961"
            },
            {
                "strMeal": "Carrot Cake",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/vrspxv1511722107.jpg",
                "idMeal": "52897"
            },
            {
                "strMeal": "Chelsea Buns",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/vqpwrv1511723001.jpg",
                "idMeal": "52898"
            },
            {
                "strMeal": "Eccles Cakes",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/wtqrqw1511639627.jpg",
                "idMeal": "52888"
            },
            {
                "strMeal": "Gigantes Plaki",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/b79r6f1585566277.jpg",
                "idMeal": "53012"
            },
            {
                "strMeal": "Lamb Tagine",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/yuwtuu1511295751.jpg",
                "idMeal": "52843"
            },
            {
                "strMeal": "Peach & Blueberry Grunt",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/ssxvup1511387476.jpg",
                "idMeal": "52862"
            },
            {
                "strMeal": "Pear Tarte Tatin",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/rxvxrr1511797671.jpg",
                "idMeal": "52916"
            },
            {
                "strMeal": "Portuguese custard tarts",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/vmz7gl1614350221.jpg",
                "idMeal": "53046"
            },
            {
                "strMeal": "Pumpkin Pie",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/usuqtp1511385394.jpg",
                "idMeal": "52857"
            },
            {
                "strMeal": "Smoky Lentil Chili with Squash",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/uwxqwy1483389553.jpg",
                "idMeal": "52784"
            },
            {
                "strMeal": "Strawberry Rhubarb Pie",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/178z5o1585514569.jpg",
                "idMeal": "53005"
            },
            {
                "strMeal": "Stuffed Lamb Tomatoes",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/u55lbp1585564013.jpg",
                "idMeal": "53008"
            },
            {
                "strMeal": "Tourtiere",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/ytpstt1511814614.jpg",
                "idMeal": "52926"
            },
            {
                "strMeal": "Venetian Duck Ragu",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/qvrwpt1511181864.jpg",
                "idMeal": "52838"
            },
            {
                "strMeal": "Walnut Roll Gužvara",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/u9l7k81628771647.jpg",
                "idMeal": "53062"
            }
        ],
        [
            {
                "strMeal": "Chicken Handi",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/wyxwsp1486979827.jpg",
                "idMeal": "52795"
            },
            {
                "strMeal": "Fish fofos",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/a15wsa1614349126.jpg",
                "idMeal": "53043"
            },
            {
                "strMeal": "Matar Paneer",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/xxpqsy1511452222.jpg",
                "idMeal": "52865"
            },
            {
                "strMeal": "Mbuzi Choma (Roasted Goat)",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/cuio7s1555492979.jpg",
                "idMeal": "52968"
            },
            {
                "strMeal": "McSinghs Scotch pie",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/vssrtx1511557680.jpg",
                "idMeal": "52880"
            }
        ],
        [
            {
                "strMeal": "Apam balik",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/adxcbq1619787919.jpg",
                "idMeal": "53049"
            },
            {
                "strMeal": "Cabbage Soup (Shchi)",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/60oc3k1699009846.jpg",
                "idMeal": "53077"
            },
            {
                "strMeal": "Feteer Meshaltet",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/9f4z6v1598734293.jpg",
                "idMeal": "53030"
            },
            {
                "strMeal": "Salmon Eggs Eggs Benedict",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/1550440197.jpg",
                "idMeal": "52962"
            }
        ],
        [
            {
                "strMeal": "Summer Pistou",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/rqtxvr1511792990.jpg",
                "idMeal": "52911"
            },
            {
                "strMeal": "Tahini Lentils",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/vpxyqt1511464175.jpg",
                "idMeal": "52869"
            },
            {
                "strMeal": "Thai Green Curry",
                "strMealThumb": "https://www.themealdb.com/images/media/meals/sstssx1487349585.jpg",
                "idMeal": "52814"
            }
        ]
    ]
}
```